features/steps/test_store_resource.py

```diff
@@ -0,0 +1,248 @@
+import os
+import shutil
+import uuid
+from dataclasses import dataclass
+from datetime import UTC, datetime
+from pathlib import Path
+from typing import Callable, Type
+
+from functools import partial
+from pytest_bdd import scenario, given, when, then
+
+from dor.adapters.bag_adapter import BagAdapter
+from dor.domain.events import (
+    Event,
+    PackageReceived,
+    PackageStored,
+    PackageSubmitted,
+    PackageVerified,
+    PackageUnpacked
+)
+from dor.providers.translocator import Translocator, Workspace
+from dor.providers.package_resource_provider import PackageResourceProvider
+from dor.service_layer.handlers.store_files import store_files
+from dor.service_layer.message_bus.memory_message_bus import MemoryMessageBus
+from dor.service_layer.unit_of_work import UnitOfWork
+from gateway.ocfl_repository_gateway import OcflRepositoryGateway
+from dor.service_layer.handlers.receive_package import receive_package
+from dor.service_layer.handlers.verify_package import verify_package
+from dor.service_layer.handlers.unpack_package import unpack_package
+from dor.providers.models import Agent, FileMetadata, FileReference, PackageResource, PreservationEvent, AlternateIdentifier, StructMap, StructMapItem, StructMapType
+
+
+class FakePackageResourceProvider:
+
+    def __init__(self, path: Path):
+        self.path = path
```

Comment:
Not sure this is the right place/time to do this, but I consider removing this `Fake` since we're not using it anymore. That'd also let you remove a good number of imports.